scanner/phase_2_missing_tracks.go

```diff
@@ -182,6 +184,61 @@ func (p *phaseMissingTracks) finalize(err error) error {
 	if matched > 0 {
 		log.Info(p.ctx, "Scanner: Found moved files", "total", matched, err)
 	}
+
+	// Check if we should purge missing items
+	if conf.Server.Scanner.PurgeMissing == "always" || (conf.Server.Scanner.PurgeMissing == "full" && p.state.fullScan) {
+		err = p.purgeMissing(err)
+		if err != nil {
+			log.Error(p.ctx, "Scanner: Error purging missing items", err)
+		}
+	}
+
+	return err
+}
+
+func (p *phaseMissingTracks) purgeMissing(err error) error {
+	count, err := p.ds.MediaFile(p.ctx).CountAll(model.QueryOptions{Filters: squirrel.Eq{"missing": true}})
+	if err != nil {
+		return fmt.Errorf("error counting missing files: %w", err)
+	}
+	
+	if count > 0 {
+		log.Info(p.ctx, "Scanner: Purging missing items from the database", "mediaFiles", count)
```

Comment:
I think this should be a method in the MediaFileRepository, something like DeleteMissing, and it should be one single SQL operation to delete all missing, instead of one delete for each missing file.

Reply:
Wasn't the transaction issue a flaky test not related to your changes? If we can't use DeleteMissing, we should create a new method, DeleteAllMissing to accomplish that.